### PR TITLE
Feat/create reveal modal

### PR DESCRIPTION
**Pull request by boranfurkan**
- Description:
- ClickUp links:
- What are the acceptance criteria? What are the steps to test that this code is working?

In this pull request I have completed the following task given in click up:
[](https://app.clickup.com/9015499537/inbox?tab=important)

I have implemented the component as like :
Screenshot 2024-05-18 at 00 59 50

I have used react-spring-bottom-sheet library as helper to speed up things. I have made local tests and seems working fine.

### CODE REVIEW

**Review by marvinmarnold — COMMENTED** [review]
@boranfurkan this looks good. I'm waiting for confirmation about how the reveal amount will be determined. 

In the meantime, can you make the currency (BONK) and burn amount be defined in global constants that can be reused in other places?

File: app/providers/RevealProvider.tsx [hunk 1]
```diff
@@ -178,20 +181,25 @@ export function RevealContextProvider({ children }: { children: ReactNode }) {
...
+              <h3 className="font-bold">Reveal answer?</h3>
+              <Button
+                onClick={() => setIsRevealModalOpen(false)}
+                className="self-end border-none w-max !p-0"
+              >
+                <CloseIcon />
+              </Button>
+            </div>
+            <p>
+              This will cost you <span className="font-bold">10000 BONK.</span>
```
Comment (marvinmarnold) [inline app/providers/RevealProvider.tsx:197]: Numbers should be displayed in this format `1,234,567.89`

**Comment by boranfurkan** [comment]
I have created costs.ts file in constants folder and added reveal cost as constant there. I have also updated the places where I could find so far. So this pr now should be ready for merging sir @marvinmarnold.

**Comment by boranfurkan** [comment]
UPDATE: Realised that I missed the formatting the currency. I have now created a function for that and implemented it. For your information @marvinmarnold.

File: app/utils/currency.ts [hunk 2]
```diff
@@ -1 +1,12 @@
 export const numberToCurrencyFormatter = new Intl.NumberFormat();
```
Comment (AnteVuletic) [inline app/utils/currency.ts:1]: This formater should be deleted and new one should be applied, or this one should be used across the project. We don't need multiple currency formaters.

**Review by marvinmarnold — APPROVED** [review]

File: app/utils/currency.ts (hunk 2 above)
Comment (marvinmarnold) [reply app/utils/currency.ts:1]: @boranfurkan bumping this comment

**Pull request merged**